src/main/java/com/example/user_management/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.example.user_management.service.impl;
+
+import com.example.user_management.entity.User;
+import com.example.user_management.repository.UserRepository;
+import com.example.user_management.service.UserService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.springframework.util.Assert.*;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+
+    @Override
+    @Transactional
+    public User createUser(final User user) {
+        assertUserNotNull(user);
+        return userRepository.save(user);
+    }
+
+    @Override
+    public Optional<User> getUserById(final Long id) {
```

Comment:
use @Transactional(readOnly=true)
Read here - https://vladmihalcea.com/read-write-read-only-transaction-routing-spring/

Reply:
Thank you